crates/uv/tests/it/init.rs

```diff
@@ -906,6 +906,10 @@ fn init_script_shebang() -> Result<()> {
 #[cfg(feature = "python-patch")]
 #[test]
 fn init_script_picks_latest_stable_version() -> Result<()> {
+    // These versions come from the ambient environment. In CI they are installed through the
+    // top-level .python-versions file, which means they come from the _released_ version of uv,
+    // not the version under test, and should only be updated _after_ a uv release and a
+    // corresponding bump to the astral-sh/setup-uv hash in CI.
```

Comment:
I don't think we should document this here. That's how `new_with_versions` works in all cases.